extension/src/client/command/FetchWorkflowSchema.ts

```diff
@@ -0,0 +1,93 @@
+/*!
+ * Copyright 2018-2019 VMware, Inc.
+ * SPDX-License-Identifier: MIT
+ */
+
+import * as path from "path"
+
+import { AutoWire, Logger, VroRestClient } from "vrealize-common"
+import * as vscode from "vscode"
+import * as fs from "fs-extra"
+
+import { Commands } from "../constants"
+import { Command } from "./Command"
+import { WorkflowNode } from "../provider/explorer/model"
+import { ConfigurationManager, EnvironmentManager } from "../manager"
+
+@AutoWire
+export class FetchWorkflowSchema extends Command {
+    private readonly logger = Logger.get("FetchWorkflowSchema")
+    private readonly restClient: VroRestClient
+
+    get commandId(): string {
+        return Commands.FetchWorkflowSchema
+    }
+
+    constructor(config: ConfigurationManager, environment: EnvironmentManager) {
+        super()
+        this.restClient = new VroRestClient(config, environment)
+    }
+
+    async execute(context: vscode.ExtensionContext, node: WorkflowNode): Promise<void> {
+        this.logger.info("Executing command Fetch Workflow Schema")
+        const storagePath = path.join(context.extensionPath, "workflow-schema")
+        if (!fs.existsSync(storagePath)) {
+            fs.mkdirpSync(storagePath)
+        }
+        const schemaFile = path.join(storagePath, `${node.id}.png`)
+        await this.fetchSchema(node, schemaFile)
+        const column = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined
+        const panel = vscode.window.createWebviewPanel(
+            "preview-workflow-schema",
+            `Workflow Schema: ${node.name}`,
+            column || vscode.ViewColumn.One,
+            {
+                localResourceRoots: [vscode.Uri.file(storagePath)]
+            }
+        )
+
+        panel.onDidDispose(() => {
+            fs.removeSync(schemaFile)
+        })
+
+        panel.webview.html = `<!DOCTYPE html>
```

Comment:
Can we extract this HTML template along with any others that we have?

Reply:
If extracted, we'll have to load it from the fs each time this command is executed. I'd rather leave it inline, it isn't a big template anyway.

Reply:
If extracted, we'll have to load it from the fs each time this command is executed. I'd rather leave it inline, it isn't a big template anyway.

Reply:
I agree.